src/super_gradients/training/models/model_factory.py

```diff
@@ -16,102 +18,115 @@
+    if not issubclass(architecture_cls, SgModule):
+        net = architecture_cls(**arch_params.to_dict(include_schema=False))
+    else:
+        if core_utils.get_param(arch_params, "num_classes"):
+            logger.warning("Passing num_classes through arch_params is deprecated and will be removed in the next version. "
+                           "Pass num_classes explicitly to models.get")
+            num_classes = arch_params.num_classes
 
-    remote_model = False
-    if isinstance(name, str) and name in ARCHITECTURES.keys():
-        architecture_cls = ARCHITECTURES[name]
-        net = architecture_cls(arch_params=arch_params)
-    elif isinstance(name, str):
-        logger.info(f'Required model {name} not found in local SuperGradients. Trying to load a model from remote deci lab')
-        deci_client = DeciClient()
-        _arch_params = deci_client.get_model_arch_params(name)
+        if num_classes is not None:
+            arch_params.override(num_classes=num_classes)
 
-        if _arch_params is not None:
-            _arch_params = hydra.utils.instantiate(_arch_params)
-            base_name = _arch_params["model_name"]
-            _arch_params = HpmStruct(**_arch_params)
-            architecture_cls = ARCHITECTURES[base_name]
-            _arch_params.override(**arch_params.to_dict())
+        if pretrained_weights is None and num_classes is None:
+            raise ValueError("num_classes or pretrained_weights must be passed to determine net's structure.")
 
-            net = architecture_cls(arch_params=_arch_params)
-            remote_model = True
-        else:
-            raise ValueError("Unsupported model name " + str(name) + ", see docs or all_architectures.py for supported nets.")
-    else:
-        raise ValueError("Unsupported model model_name " + str(name) + ", see docs or all_architectures.py for supported nets.")
-    if pretrained_weights:
-        if remote_model:
-            weights_path = deci_client.get_model_weights(name)
-            load_pretrained_weights_local(net, name, weights_path)
-        else:
-            load_pretrained_weights(net, name, pretrained_weights)
-        if num_classes_new_head != arch_params.num_classes:
-            net.replace_head(new_num_classes=num_classes_new_head)
-            arch_params.num_classes = num_classes_new_head
+        if pretrained_weights:
+            num_classes_new_head = core_utils.get_param(arch_params, "num_classes", PRETRAINED_NUM_CLASSES[pretrained_weights])
+            arch_params.num_classes = PRETRAINED_NUM_CLASSES[pretrained_weights]
 
+        # Most of the SG models work with a single params names "arch_params" of type HpmStruct, but a few take **kwargs instead
+        if "arch_params" not in get_callable_param_names(architecture_cls):
+            net = architecture_cls(**arch_params.to_dict(include_schema=False))
+        else:
+            net = architecture_cls(arch_params=arch_params)
```

Comment:
This is how I handle the case when our model does not work with arch_params.
 Is there a reason to replace:
`"arch_params" not in get_callable_param_names(architecture_cls)`
by
`get_callable_param_names(architecture_cls)[1] = "arch_params" ` ?

Reply:
no. i think this is ok